test/integration/scheduler/preemption_test.go

```diff
@@ -932,98 +933,209 @@ func TestPreemptionRaces(t *testing.T) {
 	}
-	// And .status.nominatedNodeName of the medium priority pod is cleared.
-	if err := wait.Poll(100*time.Millisecond, wait.ForeverTestTimeout, func() (bool, error) {
-		pod, err := cs.CoreV1().Pods(medPriPod.Namespace).Get(context.TODO(), medPriPod.Name, metav1.GetOptions{})
-		if err != nil {
-			t.Errorf("Error getting the medium priority pod info: %v", err)
-		}
-		if len(pod.Status.NominatedNodeName) == 0 {
-			return true, nil
-		}
-		return false, err
-	}); err != nil {
-		t.Errorf(".status.nominatedNodeName of the medium priority pod was not cleared: %v", err)
+
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			cfg := configtesting.V1beta3ToInternalWithDefaults(t, v1beta3.KubeSchedulerConfiguration{
+				Profiles: []v1beta3.KubeSchedulerProfile{{
+					SchedulerName: pointer.StringPtr(v1.DefaultSchedulerName),
+					Plugins:       tt.outOfTreePlugins,
+				}},
+			})
+			testCtx := initTest(
+				t,
+				"preemption",
+				scheduler.WithProfiles(cfg.Profiles...),
+				scheduler.WithFrameworkOutOfTreeRegistry(tt.outOfTreeRegistry),
+			)
+			t.Cleanup(func() {
+				testutils.CleanupTest(t, testCtx)
+			})
+
+			cs, ns := testCtx.ClientSet, testCtx.NS.Name
+			// Create a node with the specified capacity.
+			nodeName := "fake-node"
+			if _, err := createNode(cs, st.MakeNode().Name(nodeName).Capacity(tt.nodeCapacity).Obj()); err != nil {
+				t.Fatalf("Error creating node %v: %v", nodeName, err)
+			}
+
+			// Create pods and run post check if necessary.
+			for i, pods := range tt.podsToCreate {
+				for _, p := range pods {
+					p.Namespace = ns
+					if _, err := createPausePod(cs, p); err != nil {
+						t.Fatalf("Error creating pod %v: %v", p.Name, err)
+					}
+				}
+				// If necessary, run the post check function.
+				if len(tt.postChecks) > i && tt.postChecks[i] != nil {
+					for _, p := range pods {
+						if err := tt.postChecks[i](cs, p); err != nil {
+							t.Fatalf("Pod %v didn't pass the postChecks[%v]: %v", p.Name, i, err)
+						}
+					}
+				}
+			}
+
+			// Delete the node if necessary.
+			if tt.deleteNode {
+				if err := cs.CoreV1().Nodes().Delete(context.TODO(), nodeName, *metav1.NewDeleteOptions(0)); err != nil {
+					t.Fatalf("Node %v cannot be deleted: %v", nodeName, err)
+				}
+			}
+
+			// Force deleting the terminating pods if necessary.
+			// This is required if we demand to delete terminating Pods physically.
+			for _, podName := range tt.podNamesToDelete {
+				if err := deletePod(cs, podName, ns); err != nil {
+					t.Fatalf("Pod %v cannot be deleted: %v", podName, err)
+				}
+			}
+
+			// Verify if .status.nominatedNodeName is cleared.
+			if err := wait.Poll(100*time.Millisecond, wait.ForeverTestTimeout, func() (bool, error) {
+				pod, err := cs.CoreV1().Pods(ns).Get(context.TODO(), "medium", metav1.GetOptions{})
+				if err != nil {
+					t.Errorf("Error getting the medium pod: %v", err)
+				}
+				if len(pod.Status.NominatedNodeName) == 0 {
```

Comment:
should we have test cases for when the nnn is not cleared?

Alternatively, instead of special casing "medium", is it worth having a map of the expected nnn or assigned nodes for all the pods not removed? Then you can do a `cmp.Diff`.

Reply:
probably, but I won't bother here until we have a valid requirement in the future.

Reply:
> should we have test cases for when the nnn is not cleared?

We have e2e test covers this, like `PreemptionExecutionPath`. But I'm going to add an integration test for this specifically.

Reply:
I did add a new integration test, but it needs to consistently verify if the pod's nomiantedNode is _NOT_ cleared. It adds up the test time by 15 Seconds * 2 (the first check is to ensure an upcoming low-priority doesn't get scheduled, the second check is to ensure the preemptor's nnn is not cleared).

Reply:
Uhm... sad that there is no way to know if there was a scheduling attempt that includes all the pod/node updates.

Maybe we shouldn't commit a test that is expected to stay idle for 15 seconds?

Which leave us with an untested path, but we still have the e2e and unit tests. WDYT? Sorry for the back and forth.

Reply:
> Maybe we shouldn't commit a test that is expected to stay idle for 15 seconds?

That's my concern as well.

> Which leave us with an untested path, but we still have the e2e and unit tests. WDYT?

I think we can remove this integration test. The e2e test `PreemptionExecutionPath` has ensured the same path coverage.